hawk/src/main/java/com/orhanobut/hawk/Hawk.java

```diff
@@ -178,4 +239,97 @@ public static boolean resetCrypto() {
     public static LogLevel getLogLevel() {
         return logLevel;
     }
+
+    /**
+     * Provides the ability to chain put invocations, for example
+     * <code>Hawk.chain().put("foo", 0).put("bar", 1).done()</code>
+     */
+    public static final class HawkChain {
+
+        private final List<Pair<String, ?>> items;
+        private boolean atomic;
+
+        public HawkChain() {
+            this(4);
+        }
+
+        public HawkChain(int capacity) {
+            items = new ArrayList<>(capacity);
+        }
+
+        /**
+         * Saves every type of Objects. List, List<T>, primitives
+         *
+         * @param key   is used to save the data
+         * @param value is the data that is gonna be saved. Value can be object, list type, primitives
+         */
+        public <T> HawkChain put(String key, T value) {
+            String fullText = Hawk.put(key, value, null, false, false);
+            addItem(key, fullText);
+            return this;
+        }
+
+        /**
+         * Saves the list of objects to the storage
+         *
+         * @param key  is used to save the data
+         * @param list is the data that will be saved
+         */
+        public <T> HawkChain put(String key, List<T> list) {
+            String fullText = Hawk.put(key, null, list, true, false);
+            addItem(key, fullText);
+            return this;
+        }
+
+        /**
+         * If true, Hawk throws an exception if any of the chained invocations fails.
+         * Disabled by default.
+         *
+         * @param atomic true to throw an exception, false otherwise.
+         */
+        public HawkChain atomic(boolean atomic) {
+            this.atomic = atomic;
+            return this;
+        }
+
+        /**
+         * Saves the chained values.
+         */
+        public boolean done() {
+            return storage.put(items);
+        }
+
+        /**
+         * Saves the chained values and returns the saved keys.
+         *
+         * @return the set of saved keys.
+         */
+        public Set<String> doneWithKeys() {
```

Comment:
We should not put this option in order to avoid confusion/error. done(commit) should return false/true in order to let the user to know the result of the operation. And I also think this case won't be used quite often since Hawk does already an I/O operation which has a cost. User should put the keys and values in a map already.